### PR TITLE
#79 Fixed the startup policy

### PR DESCRIPTION
**Pull request by akara**
Components should be able to start and all signals are received until the start timeout.

### CODE REVIEW

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala [hunk 1]
```diff
@@ -495,6 +503,29 @@ class Unicomplex extends Actor with Stash with ActorLogging {
     else Active
   }
```
Comment (zhuchenwang) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: Is the only difference between lenientStrategy and failFastStrategy the sequence of "if else" statement? 
I see a lot of duplicate code. Can we use partial function to compose the strategy and eliminate duplicate code?

Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: You're right. The main difference is the sequence, which determines the outcome of the system state check. Let me try to work on composition a bit more. We should be able to get it more concise. Thanks!

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/UnicomplexBoot.scala [hunk 2]
```diff
@@ -545,7 +549,9 @@ case class UnicomplexBoot private[unicomplex] (startTime: Timestamp,
 
   def stopJVMOnExit: UnicomplexBoot = copy(stopJVM = true)
 
-  def start(): UnicomplexBoot = synchronized {
+  def start(): UnicomplexBoot = start(defaultStartupTimeout)
+
+  def start(timeout: Timeout): UnicomplexBoot = synchronized {
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/UnicomplexBoot.scala]: Codacy Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/akara-sucharitakul/squbs/file/993805146/issues/source?bid=2834325&fileBranchId=2869173#l554)

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala [hunk 3]
```diff
@@ -432,69 +440,81 @@ class Unicomplex extends Actor with Stash with ActorLogging {
...
+    case _ if !activated => Initializing
+  }
+
+  val active = { _: Any => Active }
+
+  def failFastStrategy = checkStateFailed orElse checkStateInitializing applyOrElse (cubeStates, active)
+
+  def lenientStrategy  = checkStateInitializing orElse checkStateFailed applyOrElse (cubeStates, active)
+
+  var checkInitState = lenientStrategy _
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/akara-sucharitakul/squbs/file/993805145/issues/source?bid=2834325&fileBranchId=2869173#l516)

**Pull request merged**